pkg/apis/app/v1alpha1/application_types.go

```diff
@@ -100,6 +101,34 @@ type InfoItem struct {
 
 	// Value is human readable content.
 	Value string `json:"value,omitempty"`
+
+	// Value is derived from another source.
+	ValueFrom *InfoItemSource `json:"valueFrom,omitempty"`
+}
+
+type InfoItemSource struct {
+	SecretKeyRef    *core.SecretKeySelector    `json:"secretKeyRef,omitempty"`
```

Comment:
You want need to have a type field to indicate the member of the union that contains the value. The same as InfoItem above. Each of the values needs a comment to indicate the use.

Reply:
Similar to above. I'm drawing from the following example in core:
https://github.com/kubernetes/api/blob/184e700b32b7f1b532b9fce8dd8c1f412d297c4b/core/v1/types.go#L1681